src/Tests.Microsoft.Security.Utilities.Core/WellKnownRegexPatternsTests.cs

```diff
@@ -26,6 +26,54 @@ public class WellKnownRegexPatternsTests
             "SEC101/109.AzureContainerRegistryLegacyKey"
         };
 
+        [TestMethod]
+        public void WellKnownRegexPatterns_AllRuleIdsAndNamesAreUnique()
+        {            
+            using var assertionScope = new AssertionScope();
+
+            var patterns = GetAllPatterns();
+
+            HashSet<string> ruleIds = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            HashSet<string> ruleNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            bool result;
+            foreach (RegexPattern pattern in patterns)
+            {
+                result = ruleIds.Contains(pattern.Id);
+                result.Should().BeFalse(because: $"Pattern '{pattern.GetType().Name}' should not share its Id with another rule: '{pattern.Id}'");
+
+                result = ruleNames.Contains(pattern.Name);
```

Comment:
```suggestion
                bool duplicatePatternName = ruleNames.Contains(pattern.Name);
```

nit: instead of using generic `result` everywhere some more specific names will increase readability. #WontFix

Reply:
I don't do this. I set result to my test condition so that it's on one line. Then I follow it with an assertion with a strong readable message.

`result` isn't intended to encode something useful, the detailed, highly authored message it. 

i *could* allocate new variables with names but this tends to consume more space and i worry about readability.

let's punt for now i will think about your suggestion.

Reply:
I read the code top to bottom. A well-named variable means I know how to read/check the line without the out-of-order flow.

Better yet if you want the `because` to be the sole communicator, skip using the variable altogether:

``` csharp
alreadySeeanRullName.Should().Not().Contains(pattern.Name, because: "…");
```

These are nits, so keep thinking on it.

Reply:
It is a good suggestion to encode more information about the test expectation in the variable name, I will keep this in mind moving forward.